src/bin/smbios-battery-ctl

```diff
@@ -78,6 +109,88 @@ def print_express_charge_status(n):
     else : print('\t Unknown')
 
 
+class ChargingMode(Enum):
+    PRIMARILY_AC = 0x0341
+    ADAPTIVE = 0x0342
+    CUSTOM = 0x0343  # 0x0348 is deprecated, cf. token_list.csv
+    STANDARD = 0x0346
+    EXPRESS = 0x0347
+
+    def __str__(self) -> str:
+        return self.name.lower()
+
+    @staticmethod
+    def from_string(s: str) -> 'ChargingMode':
+        try:
+            return ChargingMode[s.upper()]
+        except KeyError:
+            raise ValueError()
+
+
+CUSTOM_CHARGE_START = 0x0349
+CUSTOM_CHARGE_END = 0x034A
+
+
+def get_charging_mode() -> ChargingMode:
+    table = smbios_token.TokenTable()
+    active_mode = None
+    for mode in ChargingMode:
+        active = table[mode.value].isActive()
+        if active:
+            if active_mode is not None:
+                print(f"\t WARNING: Inconsistency: both {active_mode} and\
+                     {mode} enabled at the same time")
```

Comment:
Have you seen this happen?  I would think this should be an `Exception` actually.

Reply:
That's a good idea.